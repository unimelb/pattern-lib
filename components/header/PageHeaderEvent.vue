<template>
  <div class="page-header-event">
    <div class="page-header-event__inner">
      <h4
        v-if="headerText"
        class="page-header-event__header">
        {{ headerText }}
      </h4>
      <div class="page-header-event__body">
        <h2 class="page-header-event__title">
          {{ titleText }}
        </h2>
        <h4
          v-if="subTitleText"
          class="page-header-event__sub-title">
          {{ subTitleText }}
        </h4>
        <p
          v-if="descriptionText"
          class="page-header-event__description">
          {{ descriptionText }}
        </p>
      </div>
      <div class="page-header-event__footer">
        <ButtonIcon
          v-if="buttonLink"
          :href="buttonLink"
          class="page-header-event__button btn--event"
          size="xsml"
          width="fullwidth"
          no-icon>
          {{ buttonText }}
        </ButtonIcon>
        <SocialListGeneric
          facebook-link="www.unimelb.edu.au"
          twitter-link="www.unimelb.edu.au"
          linkedin-link="www.unimelb.edu.au"
          instagram-link="www.unimelb.edu.au"
          vimeo-link="www.unimelb.edu.au"
          youtube-link="www.unimelb.edu.au" />
      </div>
    </div>
  </div>
</template>

<script>
import SocialListGeneric from '../social-list/SocialListGeneric.vue';

export default {
  components: {
    SocialListGeneric,
  },
  props: {
    headerText: {
      type: String,
      default: '',
    },
    titleText: {
      type: String,
      default: '',
      required: true,
    },
    subTitleText: {
      type: String,
      default: '',
    },
    descriptionText: {
      type: String,
      default: '',
    },
    buttonText: {
      type: String,
      default: '',
    },
    buttonLink: {
      type: String,
      default: '',
    },
  },
};
</script>
