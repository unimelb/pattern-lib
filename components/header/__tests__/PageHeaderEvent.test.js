import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import PageHeaderEvent from '../PageHeaderEvent.vue';

expect.extend(toHaveNoViolations);

describe('PageHeaderEvent', () => {
  it('should match snapshot', () => {
    const result = shallow(PageHeaderEvent, {
      propsData: {
        titleText: 'title text',
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(PageHeaderEvent, {
      propsData: {
        titleText: 'mock title text',
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
