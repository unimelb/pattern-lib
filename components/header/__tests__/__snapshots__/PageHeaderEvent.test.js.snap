// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeaderEvent should match snapshot 1`] = `
<div
  class="page-header-event"
>
  <div
    class="page-header-event__inner"
  >
    <!---->
     
    <div
      class="page-header-event__body"
    >
      <h2
        class="page-header-event__title"
      >
        
        title text
      
      </h2>
       
      <!---->
       
      <!---->
    </div>
     
    <div
      class="page-header-event__footer"
    >
      <!---->
       
      <!---->
    </div>
  </div>
</div>
`;
